contracts/BodhiToken.sol

```diff
@@ -84,19 +92,23 @@ contract BodhiToken is StandardToken, Ownable {
   }
 
   /// @notice Allows contract owner to mint tokens at any time
-  /// @param _amount Amount of tokens to mint
+  /// @param _amount Amount of tokens to mint in lowest denomination of BOT
   function mintReservedTokens(uint256 _amount) onlyOwner {
     uint256 checkedSupply = totalSupply.add(_amount);
     require(checkedSupply <= tokenTotalSupply);
 
     mint(owner, _amount);
   }
 
-  /// @notice Shows the Ether to BOT exchange rate
-  /// @param _weiAmount Ether amount to convert
+  /// @notice Shows the amount of BOT the user will receive for amount of exchanged wei
+  /// @param _weiAmount Exchanged wei amount to convert
   /// @return The amount of BOT that will be received
   function getTokenExchangeAmount(uint256 _weiAmount) constant returns(uint256) {
-    return initialExchangeRate.mul(_weiAmount);
+    require(_weiAmount > 0);
+    assert(exchangeTokenDecimals >= decimals);
```

Comment:
we should have `assert(exchangeTokenDecimals >= decimals);` in constructor, instead of here

Reply:
good point, should fail on contract creation before its too late